src/v/wasm/cache.cc

```diff
@@ -283,9 +289,31 @@ ss::future<> caching_runtime::start(runtime::config c) {
     co_await _underlying->start(c);
     co_await _engine_caches.start();
     _gc_timer.arm(_gc_interval);
+
+    namespace sm = ss::metrics;
+    _public_metrics.add_group(
+      prometheus_sanitize::metrics_name("wasm_binaries"),
+      {
+        sm::make_gauge(
+          "executable_memory_usage",
+          sm::description(
+            "The amount of executable memory used for WebAssembly binaries"),
+          [this]() {
+              uint64_t memory_usage = 0;
+              for (const auto& [_, factory] : _factory_cache) {
+                  if (!factory) {
+                      continue;
+                  }
+                  memory_usage += factory->executable_memory_size_bytes();
+              }
+              return memory_usage;
+          })
```

Comment:
guessing the answer is "no", but is there any way to perform this aggregation outside the metric handler e.g. as items are added to the factory cache?

Reply:
We certainly could, but that begs how do we clean this up? We'd have to undo the aggregated value upon deletion, which would mean change the weak_ptr that we use in the cache. We could save raw pointers in the map, then have a custom destructor for the factory that runs and deletes the entry pointer from the map. Do note that there will only be a single factory per function in a process.

Thoughts?

Reply:
Hmm, yea i see. Seems like something to keep in mind but probably not a blocker? 

How many functions would you expect to find in a given process? My vague understanding is that these metric endpoints get hit "very frequently" but I don't have a clear sense of that numerically.

Reply:
The default limit is 16 and I expect very few right now. FWIW we are going to have to iterate over all the engines for each core and compute the memory usage too, which is dynamic and we'll have to iterate over.


Metric scapes are every 15 seconds by default in prometheus if I remember correctly

Reply:
This is a very valid concern yes (and ties in with my comment on the other PR). 

Metric scrape interval depends a bit on cluster etc. The problem isn't generally the continuous CPU load but rather the spike. In large metric series scenarios (i.e.: large partition counts) we can see reactor stalls in the milliseconds. It's a "death by a thousand cuts" kind of thing. One probe is fine but if we could slice the overall time in half that would actually make a big difference.

Reply:
Thanks - I'll respond on the other PR, but I've updated this to be a precomputed guage.